src/stream/src/executor/chain.rs

```diff
@@ -184,11 +184,19 @@ mod test {
         let chain = ChainExecutor::new(first, second, vec![0], notifier, schema);
 
         let mut chain = Box::new(chain).execute();
+        chain
+            .next()
+            .await
+            .transpose()
+            .unwrap()
+            .unwrap()
+            .as_barrier()
+            .expect("the first message received by chain must be a barrier");
 
-        let mut count = 0;
+        let mut count = 3;
```

Comment:
The chain executor will consume the `first` executor at the beginning. So this should be kept `0`?

Reply:
The first barrier mutation is none, so the chain executor just consume the second executor.

Reply:
Get it. I believe we intended to expect a barrier with mutation in this test... Could you please help to fix this?